refactor(core): backToCart API

milesap opened the pull request:
## Description
Closes #1565

## Testing
:eye: 

## Documentation
inline

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] commits messages follow the guidelines
- [x] code passes unit tests
- [ ] release notes have been updated
- [ ] PR targets the correct release version

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1602)
<!-- Reviewable:end -->

james-rae commented:
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/index-fgp-en.html, line 563 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KaiA94SGInuBOpr7qxh:-KaiA94SGInuBOpr7qxi:b-k5yl5l) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/d8745f20d531a517b11312315ff68778d29e7faa/src/index-fgp-en.html#L563)):*
> ```HTML
>                 const bookmark = RV.getMap('fgpmap')
>                     .getBookmark()
>                     .then(b => sessionStorage.setItem('fgpmap', b));
> ```

Arrow function in html makes IE cry

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

james-rae commented:
You probably want to apply the same fix to `index-fgp-fr.html`

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-KaiC9lIF27E3W0hf6Cy:b-xnihnl)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/index-fgp-en.html, line 561 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kai1VWcha6IT183AgI9:-Kai1VWcha6IT183AgIA:bjmkioc) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/d8745f20d531a517b11312315ff68778d29e7faa/src/index-fgp-en.html#L561)):*
> ```HTML
>             .setAction(function () {
>                 // save bookmark for restoration on user return
>                 const bookmark = RV.getMap('fgpmap')
> ```

I think we need to sort out something here instead of hard coding the map ID in here.

Also, is there a possible race condition between `getBookmark` and `getRcsLayerIDs`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Made changes to `index-fgp-fr.html` :+1: 

---

Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/index-fgp-en.html, line 561 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kai1VWcha6IT183AgI9:-KaladwatkQd2f3dZ32z:b64ep56) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/d8745f20d531a517b11312315ff68778d29e7faa/src/index-fgp-en.html#L561)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I think we need to sort out something here instead of hard coding the map ID in here.

Also, is there a possible race condition between `getBookmark` and `getRcsLayerIDs`?
</blockquote></details>

I can see this as an issue for third party plugins, but then again their plugin could take the map ID as a constructor parameter or special setter method. Any ideas for this?  I nestled the `getRcsLayerIDs` in `getBookmark` to avoid a race condition if there was one.

---

*[src/index-fgp-en.html, line 563 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KaiA94SGInuBOpr7qxh:-Kal_-wjtHqZnc72RM_z:b-ac4n1v) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/d8745f20d531a517b11312315ff68778d29e7faa/src/index-fgp-en.html#L563)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

Arrow function in html makes IE cry
</blockquote></details>

Ah yes, fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-KalaXkSQecTKRCh7mGW:b-6a36sw)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 2 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
- Base plugins (like MenuItem) now reside in `global-registry.js`
- Core plugins that use base plugins (like BackToCart) are now in `src/plugins/core`
- When registering a plugin you must now pass in a plugin constructor function instead of a plugin instance since core plugins are not available immediately
- Plugins can use `this.api` which contains all `apiBlock` methods in `core.run.js`

---

Review status: 1 of 11 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-Kas-kGgY03kknQAnIW7:b-7vjgzk)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-Kas22esNDTMdDmTG1uA:b-prixxw) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
> ```JavaScript
>         function register(plugin, api) {
>             // initalize the plugin and give it api scope
>             plugin = plugin();
> ```

is this affecting the function parameter?  or are we just hijacking the function parameter and overwriting the local pointer (i.e. saving us from typing `const p = plugin();`)?

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-Kas2j7sBLVy6utGG08U:b-v1e42y) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
> ```JavaScript
> 
>         // Add any MenuItem plugins as they are created to the menu
>         pluginService.onCreate('BackToCart', mItem => {
> ```

Does this mean we always create a Back to Cart menu option?  Or it only reacts if we happen to have a Back to Cart?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-Kas4HCYvCNCymuEVDwC:bjruhj) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

is this affecting the function parameter?  or are we just hijacking the function parameter and overwriting the local pointer (i.e. saving us from typing `const p = plugin();`)?
</blockquote></details>

This is a laziness hijacking :gun: 

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-Kas3O6d694fWXTPSc2Q:b9ueboo) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, james-rae (James Rae) wrote…</i></summary><blockquote>

Does this mean we always create a Back to Cart menu option?  Or it only reacts if we happen to have a Back to Cart?
</blockquote></details>

This means "Run the callback once for every `BackToCart` plugin already created and for any created later on (if any)." When run the function always adds a menu option for each `BackToCart` plugin.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 2 files at r2, 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[gulp.config.js, line 55 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2zkV_Ab1m_XdtIRm:-Kas2zkV_Ab1m_XdtIRn:b-i93ilk) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/gulp.config.js#L55)):*
> ```JavaScript
>             'templates.js',
>             'app-seed.js',
>             'plugins.js',
> ```

This should be `corePlugins.js` (I think we were just lucky it merged into an order that worked when we tested).

---

*[src/index-fgp-fr.html, line 581 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3LEZrgmJ8e8L0-WR:-Kas3LE_8Y3CTN31GEOx:b-bplrp9) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/index-fgp-fr.html#L581)):*
> ```HTML
>         }
>         function testBackToCart(){
>             RV.getMap('fgpmap').getBookmark().then(function (b) {
> ```

Can you copy the changes over from `fgp-en` ?

---

*[src/app/global-registry.js, line 45 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3aniD3xLuNyxMzTW:-Kas3aniD3xLuNyxMzTX:bbbs17i) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/global-registry.js#L45)):*
> ```JavaScript
>         set id (id) { this._id = id; }
> 
>         set api (api) { this._api = api; }
> ```

Is it possible to work out a solution that doesn't allow `api` to be writable (i.e. can we bind it within the scope of `global-registry` but then not have it leak out past that module?

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-KasBDSIJIWucdvDewWz:b-mp548r) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

This is a laziness hijacking :gun: 
</blockquote></details>

If I had a muffin I might be willing to overlook this.

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-Kas4hq44rFHJOSQpMvg:b-hrxil5) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

This means "Run the callback once for every `BackToCart` plugin already created and for any created later on (if any)." When run the function always adds a menu option for each `BackToCart` plugin.
</blockquote></details>

Although `BackToCart` is a core plugin we probably don't want to design such that the app knows about the plugins unless it's important (e.g. the geolocation example).

I'm not sure I follow why this had to change to a more specific initialization.

---

*[src/plugins/core/back-to-cart.js, line 16 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas67JY5opus-ZghkxC:-Kas67JZkujSZ3fnrFOh:bfj6qrs) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/plugins/core/back-to-cart.js#L16)):*
> ```JavaScript
>     class BackToCart extends RV.BasePlugins.MenuItem {
> 
>         set catalogURL (url) { this._catalogURL = url.replace(/\/$/, ''); }
> ```

Instead of taking the base URL and assuming no other parameters I think we should offer a format string option.  They pass something like `http://cart.example.com/layers?lang=en&keys={RV_LAYER_LIST}` and we do a substitution on the parameters we recognize (just layer list for now, but we could add more if it's useful).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-KasDk21asXYksbrwmzk:bufgv38) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

If I had a muffin I might be willing to overlook this.
</blockquote></details>

King muffin is yours then

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KasD1LxFek7Dd7WHETZ:btvnwyu) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Although `BackToCart` is a core plugin we probably don't want to design such that the app knows about the plugins unless it's important (e.g. the geolocation example).

I'm not sure I follow why this had to change to a more specific initialization.
</blockquote></details>

`RV.Plugins` is not available here, it is `undefined` at this point. I agree we shouldn't use BackToCart though since other plugins can extend `MenuItem`. I could possible change this to something like `pluginService.onCreate('MenuItem', mItem => {`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

alyec commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-KasExwH-S4akmEJQAVC:b8dfd92) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
> *might*

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KasEaonOjV3qgbMrpdl:b-kdxhbc) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

`RV.Plugins` is not available here, it is `undefined` at this point. I agree we shouldn't use BackToCart though since other plugins can extend `MenuItem`. I could possible change this to something like `pluginService.onCreate('MenuItem', mItem => {`.
</blockquote></details>

`'MenuItem'` works for me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

jolevesq commented:
Reviewed 1 of 2 files at r2, 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 4 of 10 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[src/index-fgp-en.html, line 556 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KavtiphUcM-IP6Z5DwU:-KavtiphUcM-IP6Z5DwV:bq5bhgx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/index-fgp-en.html#L556)):*
> ```HTML
> 
>         RV.getMap('fgpmap').registerPlugin(
>             () => new RV.Plugins.BackToCart('backToCart', window.location.href.split('?')[0])
> ```

Arrow functions in index files -> ![](http://forum.the-big-bang-theory.com/uploads/emoticons/default_resent.gif) ![](https://online.marsh.com/tspv3public/TSPServices/images/ie-alert-icn.png)

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-Kavv-XZOu6ZfyoE7387:b-c38jjs) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
> ```JavaScript
>         },
> 
>         registerPlugin(plugin) {
> ```

jsdoc is missing.

---

*[src/app/core/core.run.js, line 170 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavw6aHAOoKCEpkXuCN:-Kavw6aHAOoKCEpkXuCO:b-s4ei8f) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/core.run.js#L170)):*
> ```JavaScript
> 
>         // need to pass service, does not work with 'this' inside {}
>         service.registerPlugin = plugin => pluginService.register(plugin, service);
> ```

This should work:
```js
const service = {
	...
	appInfo,
	registerPlugin(plugin) {
		pluginService.register(plugin, this);
	}
};
```

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-KavwhOb8R8anvLMzsMi:b-p4cpdy) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

> *might*
</blockquote></details>

I want a muffin too.

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KavxQaeBeGgc47I21VN:b-lrlkgz) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`'MenuItem'` works for me.
</blockquote></details>

Would this fire when the language is changed? Switching language rebuilds the side menu, so I feel this belongs to `setupSidenavButtons`. Also, `SIDENAV_CONFIG_DEFAULT` should not be modified; it's used as the default (that's why it's outside the service function). `service.config` needs to be modified instead.

---

*[src/plugins/core/back-to-cart.js, line 19 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KavykI0iUlSqWWJsWX9:-KavykI0iUlSqWWJsWXA:b-c38jjs) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/plugins/core/back-to-cart.js#L19)):*
> ```JavaScript
>         get catalogURL () { return this._catalogURL; }
> 
>         onMenuItemClick () {
> ```

jsdoc is missing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

alyec commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KbAV6V88uiVMgmRBPLH:b-a4q0l5) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Would this fire when the language is changed? Switching language rebuilds the side menu, so I feel this belongs to `setupSidenavButtons`. Also, `SIDENAV_CONFIG_DEFAULT` should not be modified; it's used as the default (that's why it's outside the service function). `service.config` needs to be modified instead.
</blockquote></details>

This gets a bit more tricky.  Some plugins may want to retain state across proj switches / lang switch / navigation; I think we'll need to expand the plugin API a bit.

For the moment the immediate plugins are essentially stateless so I think we're okay with a destroy / re-create cycle.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 2 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[gulp.config.js, line 55 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2zkV_Ab1m_XdtIRm:-KavrkDaPDIK7mkaNC63:b-a2ujgl) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/gulp.config.js#L55)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

This should be `corePlugins.js` (I think we were just lucky it merged into an order that worked when we tested).
</blockquote></details>

Changed

---

*[src/index-fgp-en.html, line 556 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KavtiphUcM-IP6Z5DwU:-KbBh48GOD1hz4M4Fj6t:bq97wgm) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/index-fgp-en.html#L556)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Arrow functions in index files -> ![](http://forum.the-big-bang-theory.com/uploads/emoticons/default_resent.gif) ![](https://online.marsh.com/tspv3public/TSPServices/images/ie-alert-icn.png)
</blockquote></details>

I try to sneak them in to put IE out of it's misery. Didn't work this time

---

*[src/index-fgp-fr.html, line 581 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3LEZrgmJ8e8L0-WR:-KbBhYw5KKp6NgokSiBh:b-8fx2u7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/index-fgp-fr.html#L581)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Can you copy the changes over from `fgp-en` ?
</blockquote></details>

:+1: 

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-KbBjDms-H_TmDNjXgmy:b8zjx54) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

jsdoc is missing.
</blockquote></details>

They're all missing :worried: 

---

*[src/app/global-registry.js, line 45 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3aniD3xLuNyxMzTW:-KbBjlGfF5cxnb9P4b2o:b8xbr4i) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/global-registry.js#L45)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Is it possible to work out a solution that doesn't allow `api` to be writable (i.e. can we bind it within the scope of `global-registry` but then not have it leak out past that module?
</blockquote></details>

An error is thrown if api is set, let me know if this is ok. I assume we can't stop them from changing `_api`

---

*[src/app/core/core.run.js, line 170 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavw6aHAOoKCEpkXuCN:-KbBjzXhIzsz1TOZ-U_9:b-8fx2u7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/core.run.js#L170)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

This should work:
```js
const service = {
	...
	appInfo,
	registerPlugin(plugin) {
		pluginService.register(plugin, this);
	}
};
```
</blockquote></details>

:+1: 

---

*[src/app/core/plugin.service.js, line 36 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas22er_Zveb-xU2m4a:-KavtaICSot1Qe0FNFeq:b-1eatxy) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/core/plugin.service.js#L36)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

I want a muffin too.
</blockquote></details>

I see. To avoid fighting over king muffin I made the change.

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KbFjiDWwurZZ8VU8kWS:b-xbkl7d) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

This gets a bit more tricky.  Some plugins may want to retain state across proj switches / lang switch / navigation; I think we'll need to expand the plugin API a bit.

For the moment the immediate plugins are essentially stateless so I think we're okay with a destroy / re-create cycle.
</blockquote></details>

This does work when language is switched. Let me know if we still need this changed. 

Also, changed so that it is flexible in handling plugin type detection. Using `MenuItem` now but it could also accept `BackToCart.

---

*[src/plugins/core/back-to-cart.js, line 16 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas67JY5opus-ZghkxC:-Kavuq67OdTQuIf6tcpL:bs69k41) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/plugins/core/back-to-cart.js#L16)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Instead of taking the base URL and assuming no other parameters I think we should offer a format string option.  They pass something like `http://cart.example.com/layers?lang=en&keys={RV_LAYER_LIST}` and we do a substitution on the parameters we recognize (just layer list for now, but we could add more if it's useful).
</blockquote></details>

Great idea, changed

---

*[src/plugins/core/back-to-cart.js, line 19 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KavykI0iUlSqWWJsWX9:-KbFkgPQiXja1sYFge9F:b-hhti52) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/plugins/core/back-to-cart.js#L19)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

jsdoc is missing.
</blockquote></details>

Added jsdoc

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 9 of 9 files at r4.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-KbFu74bNzSWr3uZD-oA:bvmocqa) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

They're all missing :worried: 
</blockquote></details>

Can you add an issue to fix that at some point.  Especially since this is the external API we should be paying extra special attention to the docs here.

---

*[src/app/global-registry.js, line 45 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3aniD3xLuNyxMzTW:-KbFogfN0_z4zdN4fOgH:b6klwps) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/global-registry.js#L45)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

An error is thrown if api is set, let me know if this is ok. I assume we can't stop them from changing `_api`
</blockquote></details>

I don't think you need this line.  I'm pretty sure in javascript strict mode this will throw an error anyways if the `set` part of a property is not defined.  I think it fails silently in normal mode but all of our babel translated code is running in strict mode.

Regarding `_api` yes, they can do whatever they want.  There are ways around that too, but as far as I'm concerned if you're going to mess around with internal variables you get what you deserve.

---

*[src/app/core/plugin.service.js, line 22 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFtG4UASekx07KxtA2:-KbFtG4UASekx07KxtA3:bqlbweq) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L22)):*
> ```JavaScript
>         };
> 
>         const pluginList = [];
> ```

Can you add some notes on both of these?  Particularly for `createList` can you describe the contents as `{class, callback}` pairs?

---

*[src/app/core/plugin.service.js, line 37 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFs8hWp6VImfEWBk2v:-KbFs8hWp6VImfEWBk2w:by78ww0) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L37)):*
> ```JavaScript
>             // initalize the plugin and give it api scope
>             const params = [...arguments];
>             const Plugin = params.splice(0, 1)[0];
> ```

Can you add a few notes for later that args should be structured as `[ PluginClass, id:string, ...pluginInitParams, mapApiReference]`?

Also, can you add some summary of the discussion why we have `init` instead of overriding the constructor?

---

*[src/app/core/plugin.service.js, line 64 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFtcSWLilUO0jq8wAR:-KbFtcSWLilUO0jq8wAS:b-abubbw) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L64)):*
> ```JavaScript
> 
>             // execute onCreate callbacks for this plugin type
>             onCreateList.forEach(x => p instanceof x.pluginType ? x.cb(p) : null);
> ```

styling: `.filter(p instanceof type).forEach(cb)`

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KbFrN6V-UzLlIP6IWF3:bey4yxe) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

This does work when language is switched. Let me know if we still need this changed. 

Also, changed so that it is flexible in handling plugin type detection. Using `MenuItem` now but it could also accept `BackToCart.
</blockquote></details>

Some notes from a pod discussion.  Probably makes sense to keep `init` and `onCreate` semantics as on creation only.  We can subsequently add `onLangSwitch` or `onMapReload` as plugins need to be more aware of page changes.

---

*[src/app/ui/sidenav/sidenav.service.js, line 1 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFrmRARoLOZ7X4PQxA:-KbFrmRARoLOZ7X4PQxB:blinlnj) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/ui/sidenav/sidenav.service.js#L1)):*
> ```JavaScript
> /* global RV */
> ```

Should we be using `globalRegistry` instead of the declared global?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 8 of 9 files at r4.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: 8 of 11 files reviewed at latest revision, 11 unresolved discussions.

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-KbFvlTluGB1LgpwpHEn:b-hp9qbq) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Can you add an issue to fix that at some point.  Especially since this is the external API we should be paying extra special attention to the docs here.
</blockquote></details>

:+1: Issue is #1622

---

*[src/app/global-registry.js, line 45 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas3aniD3xLuNyxMzTW:-KbFvuaj5KtGJ7SypkKt:b-hduc4l) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/global-registry.js#L45)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I don't think you need this line.  I'm pretty sure in javascript strict mode this will throw an error anyways if the `set` part of a property is not defined.  I think it fails silently in normal mode but all of our babel translated code is running in strict mode.

Regarding `_api` yes, they can do whatever they want.  There are ways around that too, but as far as I'm concerned if you're going to mess around with internal variables you get what you deserve.
</blockquote></details>

Even better, removed

---

*[src/app/core/plugin.service.js, line 22 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFtG4UASekx07KxtA2:-KbFx7Tng8cKUM-lWysI:b7wcq5t) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L22)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Can you add some notes on both of these?  Particularly for `createList` can you describe the contents as `{class, callback}` pairs?
</blockquote></details>

Updated inline comments

---

*[src/app/core/plugin.service.js, line 37 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFs8hWp6VImfEWBk2v:-KbFzxvleHkjMSFjpL49:b-8fx2u7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L37)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Can you add a few notes for later that args should be structured as `[ PluginClass, id:string, ...pluginInitParams, mapApiReference]`?

Also, can you add some summary of the discussion why we have `init` instead of overriding the constructor?
</blockquote></details>

:+1: 

---

*[src/app/core/plugin.service.js, line 64 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFtcSWLilUO0jq8wAR:-KbG-GXCJU_j-tcdnyxV:b499q7j) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/core/plugin.service.js#L64)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

styling: `.filter(p instanceof type).forEach(cb)`
</blockquote></details>

Changed

---

*[src/app/ui/sidenav/sidenav.service.js, line 154 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kas2j7sBLVy6utGG08T:-KbG-fcxroxW0sPiNGYh:b-9b4d3l) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/ui/sidenav/sidenav.service.js#L154)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Some notes from a pod discussion.  Probably makes sense to keep `init` and `onCreate` semantics as on creation only.  We can subsequently add `onLangSwitch` or `onMapReload` as plugins need to be more aware of page changes.
</blockquote></details>

Agreed for future releases

---

*[src/app/ui/sidenav/sidenav.service.js, line 1 at r4](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbFrmRARoLOZ7X4PQxA:-KbG-YdqcOUaU60JP8xL:b8enhkz) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/1d7fae4ee801c68fd2bbdd7c14e0712c26916d56/src/app/ui/sidenav/sidenav.service.js#L1)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Should we be using `globalRegistry` instead of the declared global?
</blockquote></details>

Agreed, changed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 6 of 9 files at r4.
Review status: 8 of 11 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/index-fgp-en.html, line 587 at r5](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbGkRHePfVx1BqDb9Yz:-KbGkPdQ_LeMysbBOMEZ:bohmbo) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/66ff4f80f7011dda443707f4307e66c5b5c89dd6/src/index-fgp-en.html#L587)):*
> ```HTML
>         }
>         function testBackToCart(){
>             RV.getMap('fgpmap').getBookmark().then(function (bookmark) {
> ```

Can you put `RV.getMap('fgpmap')` into a variable? This is supposed to be a sample code other should follow...

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-KbGlqGn3D49r-bnMPr8:bqyy04k) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
<details><summary><i>Previously, milesap (Miles Petrov) wrote…</i></summary><blockquote>

:+1: Issue is #1622
</blockquote></details>

jsdoc wanted; 2 muffin reward.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/index-fgp-en.html, line 587 at r5](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbGkRHePfVx1BqDb9Yz:-KbGot1EC9Wt7eWHSueV:b-vj5r1q) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/66ff4f80f7011dda443707f4307e66c5b5c89dd6/src/index-fgp-en.html#L587)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Can you put `RV.getMap('fgpmap')` into a variable? This is supposed to be a sample code other should follow...
</blockquote></details>

We should flatten this promise chain as well in that case.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 9 files at r4, 3 of 3 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[src/index-fgp-en.html, line 587 at r5](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-KbGkRHePfVx1BqDb9Yz:-KbL9azmuTV49bHV_p7M:b-7oca7m) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/66ff4f80f7011dda443707f4307e66c5b5c89dd6/src/index-fgp-en.html#L587)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

We should flatten this promise chain as well in that case.
</blockquote></details>

Flattened and variablized

---

*[src/app/bootstrap.js, line 136 at r3](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-Kavv-XZOu6ZfyoE7386:-KbLCB2BSkQSNhDbu_z9:biidv6k) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/0c535a23543f37e718564390a5192060c6ac96c7/src/app/bootstrap.js#L136)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

jsdoc wanted; 2 muffin reward.
</blockquote></details>

Can't say no to muffins - jsdoc written :+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Reviewed 3 of 3 files at r5, 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->

milesap commented:
> UglifyJS does not currently support EcmaScript 6 structures like classes.

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-KbQL0l9FggcfEXek8G4:b-en30cz)*
<!-- Sent from Reviewable.io -->

milesap commented:
Prod build now working :+1: 

---

Review status: 10 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-KbQQz3_PAEp4ifFlkB7:b-e8keix)*
<!-- Sent from Reviewable.io -->

alyec commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602#-:-KbQRi64_sE7C-ny0pma:bnfp4nl)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 1 files at r7.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1602)*
<!-- Sent from Reviewable.io -->
